vault/hcp_link/proto/meta/meta.proto

```diff
@@ -34,6 +34,57 @@ message ListAuthResponse {
   repeated Auth Auths = 1;
 }
 
+message GetClusterStatusRequest {}
+
+message HANode {
+  string Hostname = 1;
+}
+
+message RaftServer {
+  // NodeID is the name of the server
+  string NodeID = 1;
+
+  // Address is the IP:port of the server, used for Raft communications
+  string Address = 2;
+
+  // Leader is true if this server is the current cluster leader
+  bool Leader = 3;
+
+  // Protocol version is the raft protocol version used by the server
+  string ProtocolVersion = 4;
+
+  // Voter is true if this server has a vote in the cluster. This might
+  // be false if the server is staging and still coming online.
+  bool Voter = 5;
+}
+
+message RaftConfiguration {
+  repeated RaftServer Servers = 1;
+}
+
+message AutopilotServer {
+  string ID = 1;
+  bool Healthy = 2;
+}
+
+message AutopilotStatus {
+  bool Healthy = 1;
+  repeated AutopilotServer Servers = 2;
+}
+
+message RaftStatus {
+  RaftConfiguration RaftConfiguration = 1;
+  AutopilotStatus AutopilotStatus = 2;
+  repeated string QuorumWarnings = 3;
```

Comment:
Following up on my comment on the RFC: https://docs.google.com/document/d/1Q6ka8XmhZTfjxHCRIXPcfkHi2dVQCulhGPhIdbzCsBc/edit?disco=AAAAmNE8yrg

Why is it a repeated field when only one warning will be returned based on [https://github.com/hashicorp/vault/blob/ccdb62d78eb25f9b85f2cef82955bd9e8d383f17/vault/diagnose/raft_checks.go#L81-L96](https://www.google.com/url?q=https://github.com/hashicorp/vault/blob/ccdb62d78eb25f9b85f2cef82955bd9e8d383f17/vault/diagnose/raft_checks.go%23L81-L96&sa=D&source=docs&ust=1670935864050120&usg=AOvVaw10VO4qSjADCnVKeBIhen7A)?

Reply:
Yup, that's definitely true. The diagnose logic will return a single error and thus has inherent priority. It is possible to have multiple issues, however. For example, a cluster can have an even number voters but also more than 7 voters. So  it seemed useful to have a repeated field.